application/org.openjdk.jmc.greychart.ui/src/main/java/org/openjdk/jmc/greychart/ui/views/ChartComposite.java

```diff
@@ -185,22 +185,13 @@ public void refresh() {
 			zoomInAction.setEnabled(m_viewWidth > MINIMUM_WORLD_WIDTH && !m_enableUpdates);
 			((NanosXAxis) getChart().getXAxis()).setRange(m_viewEnd - m_viewWidth, m_viewEnd);
 			m_chart.setXAxis(m_chart.getXAxis());
-			if (IMMEDIATE_DRAWING && isVisible() && !getClientArea().isEmpty()) {
-				GC gc = new GC(this);
-				paint(gc);
-				if (isFocusControl()) {
-					FocusTracker.drawFocusOn(this, gc);
-				}
-				gc.dispose();
-			} else {
-				redraw();
-				/*
-				 * Explicit calls to update() should be avoided unless absolutely necessary. They
-				 * may have a negative performance impact and may cause issues on Mac OS X Cocoa
-				 * (SWT 3.6). If it is required here, there must be a justifying comment.
-				 */
-				// update();
-			}
+			redraw();
+			/*
```

Comment:
It seems we can remove the comment and commented code here - explicit calls to update should not be needed.

Reply:
Comment and commented code are present from line number 189 onwards. We do not need 'If' condition and so the 'if-else' block has been removed and the codes from 'else' block has been copied outside the block.

Reply:
> It seems we can remove the comment and commented code here - explicit calls to update should not be needed.

Comment and commented code are present from line number 189 onwards. We do not need 'If' condition and so the 'if-else' block has been removed and the codes from 'else' block has been copied outside the block.

Reply:
> It seems we can remove the comment and commented code here - explicit calls to update should not be needed.

I have removed the comment and commented code.